### ISSUE TITLE
Liberty cannot be immediately restarted after stopping with localConnector-1.0 feature on Windows with hotspot

### ISSUE THREAD

**Issue by colvinco** [OpenLiberty/open-liberty#25781]
**Describe the bug**  
From version 23.0.0.1 onwards, when calling `server start <name>` immediately after `server stop <name>` completes on a running instance, the script wrongly responds that the server is already running. This worked fine in all previous releases but now fails every time, though it only appears to affect Windows and only when using the `localConnector-1.0` feature.
The server's termination appears to be delayed, as a subsequent call to start will succeed, or waiting ~1-2 seconds before calling start will also succeed.

```
Liberty\wlp\bin>server version
Open Liberty 23.0.0.1 (1.0.73.cl230120230123-2118) on OpenJDK 64-Bit Server VM, version 11.0.11+9 (en_GB)

Liberty\wlp\bin>server stop demo && server start demo
Stopping server demo.
Server demo stopped.
Server demo is already running. ### This shouldn't happen
```

```
Liberty\wlp_13\bin>server version
Open Liberty 22.0.0.13 (1.0.72.cl221320221205-1900) on OpenJDK 64-Bit Server VM, version 11.0.11+9 (en_GB)

Liberty\wlp_13\bin>server stop demo && server start demo
Stopping server demo.
Server demo stopped.
Starting server demo.
Server demo started.
```

**Steps to Reproduce**  
On Windows, create a server with a server.xml containing the `localConnector-1.0` feature:
```
<server description="demo server">
  <featureManager>
    <feature>localConnector-1.0</feature>
  </featureManager>
</server>
```
Call `server stop demo && server start demo`

**Expected behavior**  
The server stop command should only return once the process has terminated, the server start command should succeed when called immediately after server stop completes.

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.1 to 23.0.0.6
 - Java Version: ```openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)```
 - Windows 10 22H2

**Additional context**  
Can't reproduce it on Ubuntu. The behaviour changed in 23.0.0.1 and it is still the same in 23.0.0.6, I've tested it with 22.0.0.12 and .13, which are both fine.

**Comment by jhanders34** [comment]
#regression

**Comment by jimblye** [comment]
It is working for me with the default application after adding the localConnector-1.0 feature.    Can you reproduce with the default application after adding the localConnector-1.0 ?

```
C:\temp\ol23-1\wlp\usr\servers\defaultServer>cdt ..\..\..\bin
C:\temp\ol23-1\wlp\bin>server start
Starting server defaultServer.
Server defaultServer started.

C:\temp\ol23-1\wlp\bin>server stop && server start
Stopping server defaultServer.
Server defaultServer stopped.
Starting server defaultServer.
Server defaultServer started.
```

I assume I need an app that actually uses the localConnector feature.   Need info on how to create such an app.

**Comment by Colvin-Cowie-i2** [comment]
Hi @jimblye , thanks for looking. Yes I can reproduce it with the default application, or simply by creating a server that just contains a server.xml with the feature and nothing else. It doesn't even need an actual application.
```
<server description="demo server">
  <featureManager>
    <feature>localConnector-1.0</feature>
  </featureManager>
</server>
```

I've tried it with various JDKs, I've tried it natively on Windows 10 22H2, and in Windows 11 22H2 in a hyperv VM, and it happens trivially
```
C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "11.0.19" 2023-04-18
OpenJDK Runtime Environment Temurin-11.0.19+7 (build 11.0.19+7)
OpenJDK 64-Bit Server VM Temurin-11.0.19+7 (build 11.0.19+7, mixed mode)

C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Temurin-17.0.8+7 (build 17.0.8+7)
OpenJDK 64-Bit Server VM Temurin-17.0.8+7 (build 17.0.8+7, mixed mode, sharing)

C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "11.0.19" 2023-04-18
OpenJDK Runtime Environment OpenLogic-OpenJDK (build 11.0.19+7-adhoc..jdk11u)
OpenJDK 64-Bit Server VM OpenLogic-OpenJDK (build 11.0.19+7-adhoc..jdk11u, mixed mode)

C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
java version "11.0.19" 2023-04-18 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.19+9-LTS-224)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.19+9-LTS-224, mixed mode)

...
C:\openliberty-runtime-23.0.0.6\wlp\bin>server stop & server start
Stopping server defaultServer.
Server defaultServer stopped.
Server defaultServer is already running.
```

Ah, but it doesn't happen with Semeru!

```
C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "17.0.7" 2023-04-18
IBM Semeru Runtime Open Edition 17.0.7.0 (build 17.0.7+7)
Eclipse OpenJ9 VM 17.0.7.0 (build openj9-0.38.0, JRE 17 Windows 11 amd64-64-Bit Compressed References 20230418_439 (JIT enabled, AOT enabled)
OpenJ9   - d57d05932
OMR      - 855813495
JCL      - 9d7a231edbc based on jdk-17.0.7+7)

C:\openliberty-runtime-23.0.0.6\wlp\bin>server stop & server start
Stopping server defaultServer.
Server defaultServer is not running.
Starting server defaultServer.
Server defaultServer started.
```

So if you try it with a hotspot VM, I expect you'll see it

**Comment by colvinco** [comment]
Hi @jimblye , thanks for looking. Yes I can reproduce it with the default application, or simply by creating a server that just contains a server.xml with the feature and nothing else. It doesn't even need an actual application.
```
<server description="demo server">
  <featureManager>
    <feature>localConnector-1.0</feature>
  </featureManager>
</server>
```

I've tried it with various JDKs, I've tried it natively on Windows 10 22H2, and in Windows 11 22H2 in a hyperv VM, and it happens trivially
```
C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "11.0.19" 2023-04-18
OpenJDK Runtime Environment Temurin-11.0.19+7 (build 11.0.19+7)
OpenJDK 64-Bit Server VM Temurin-11.0.19+7 (build 11.0.19+7, mixed mode)

C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Temurin-17.0.8+7 (build 17.0.8+7)
OpenJDK 64-Bit Server VM Temurin-17.0.8+7 (build 17.0.8+7, mixed mode, sharing)

C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "11.0.19" 2023-04-18
OpenJDK Runtime Environment OpenLogic-OpenJDK (build 11.0.19+7-adhoc..jdk11u)
OpenJDK 64-Bit Server VM OpenLogic-OpenJDK (build 11.0.19+7-adhoc..jdk11u, mixed mode)

C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
java version "11.0.19" 2023-04-18 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.19+9-LTS-224)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.19+9-LTS-224, mixed mode)

...
C:\openliberty-runtime-23.0.0.6\wlp\bin>server stop & server start
Stopping server defaultServer.
Server defaultServer stopped.
Server defaultServer is already running.
```

Ah, but it doesn't happen with Semeru!

```
C:\openliberty-runtime-23.0.0.6\wlp\bin>java -version
openjdk version "17.0.7" 2023-04-18
IBM Semeru Runtime Open Edition 17.0.7.0 (build 17.0.7+7)
Eclipse OpenJ9 VM 17.0.7.0 (build openj9-0.38.0, JRE 17 Windows 11 amd64-64-Bit Compressed References 20230418_439 (JIT enabled, AOT enabled)
OpenJ9   - d57d05932
OMR      - 855813495
JCL      - 9d7a231edbc based on jdk-17.0.7+7)

C:\openliberty-runtime-23.0.0.6\wlp\bin>server stop & server start
Stopping server defaultServer.
Server defaultServer is not running.
Starting server defaultServer.
Server defaultServer started.
```

So if you try it with a hotspot VM, I expect you'll see it

**Comment by jimblye** [comment]
@colvinco
When this fails and you get the "is already running." message, do you see messages in console.log similar to the following?:

```
WARNING: Illegal reflective access by com.ibm.ws.kernel.service.location.internal.Activator (file:/C:/temp/ol23-1/wlp/lib/com.ibm.ws.kernel.service_1.3.73.jar) to field javax.naming.spi.NamingManager.initctx_factory_builder
WARNING: Please consider reporting this to the maintainers of com.ibm.ws.kernel.service.location.internal.Activator
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by colvinco** [comment]
I see that with Java 11
```
Launching demo (Open Liberty 23.0.0.6/wlp-1.0.78.cl230620230612-1100) on OpenJDK 64-Bit Server VM, version 11.0.19+7 (en_US)
[AUDIT   ] CWWKE0001I: The server demo has been launched.
[AUDIT   ] CWWKF0012I: The server installed the following features: [localConnector-1.0].
[AUDIT   ] CWWKF0011I: The demo server is ready to run a smarter planet. The demo server started in 1.570 seconds.
[AUDIT   ] CWWKE0055I: Server shutdown requested on Monday, July 31, 2023 at 3:15 AM. The server demo is shutting down.
[AUDIT   ] CWWKE1100I: Waiting for up to 30 seconds for the server to quiesce.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.ibm.ws.kernel.service.location.internal.Activator (file:/C:/openliberty-runtime-23.0.0.6/wlp/lib/com.ibm.ws.kernel.service_1.3.78.jar) to field javax.naming.spi.NamingManager.initctx_factory_builder
WARNING: Please consider reporting this to the maintainers of com.ibm.ws.kernel.service.location.internal.Activator
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[AUDIT   ] CWWKE0036I: The server demo stopped after 2.285 seconds.
```
But not with Java 17
```
Launching demo (Open Liberty 23.0.0.6/wlp-1.0.78.cl230620230612-1100) on OpenJDK 64-Bit Server VM, version 17.0.8+7 (en_US)
[AUDIT   ] CWWKE0001I: The server demo has been launched.
[AUDIT   ] CWWKF0012I: The server installed the following features: [localConnector-1.0].
[AUDIT   ] CWWKF0011I: The demo server is ready to run a smarter planet. The demo server started in 0.855 seconds.
[AUDIT   ] CWWKE0055I: Server shutdown requested on Monday, July 31, 2023 at 3:16 AM. The server demo is shutting down.
[AUDIT   ] CWWKE1100I: Waiting for up to 30 seconds for the server to quiesce.
[AUDIT   ] CWWKE0036I: The server demo stopped after 1.831 seconds.

```

**Comment by jimblye** [comment]
Regression ... but ... This seems to be a timing problem in this very specific scenario.   
What I found was that the client (server stop command) **has never** been waiting for the server process to actually shut down on Windows.   I don't know of any specific change that caused this scenario to fail in 23.0.0.1, but it seems the timing changed just enough for the start to fail in this very specific environment.
I found that the code only waits for the server to stop when it can retrieve the pid, but on Windows, it seems to never be able to get the pid.
